pkg/raft/node.go

```diff
@@ -417,18 +440,19 @@ func (n *node) run() {
 			}
 		case <-n.tickc:
 			n.rn.Tick()
-		case readyc <- rd:
-			n.rn.acceptReady(rd)
-			if !n.rn.asyncStorageWrites {
-				advancec = n.advancec
-			} else {
+		case acceptReadyC <- struct{}{}:
+			// Invariant: n.rn.HasReady() == true.
+			rd = n.rn.Ready()
+			n.readyW.resp <- rd // never blocks
```

Comment:
// never blocks because of contract on (Node).Ready()

Reply:
Done.